### PR TITLE
feat: influence parameter defaults through cli flag/env

### PR DESCRIPTION
**Pull request by aaronlehmann**
Add a `--parameter-default` flag / `CODER_RICH_PARAMETER_DEFAULT` environment variable which overrides default values suggested for parameters.

This allows scripts or middleware wrapping the CLI to substitute defaults for parameter values beyond those defined at the template level. For example, Git repository/branch parameters can be given defaults based on the current checkout, or default parameter values can be parsed out of files inside the repo.

### CODE REVIEW

**Comment by aaronlehmann** [comment]
> The original title does not begin with a verb. It explains an action being carried out, rather than the changes being made.
> The correct form should start with a verb, indicating the action taken in the pull request.

"influence" is a verb :(

**Comment by aaronlehmann** [comment]
Hi @mafredri, would you be the right person to review this one? Thanks in advance!

**Review by kylecarbs — APPROVED** [review]
Looks good to me!

**Review by johnstcn — APPROVED** [review]

File: cli/cliui/parameter.go [hunk 1]
```diff
@@ -10,7 +10,7 @@ import (
 	"github.com/coder/serpent"
 )
 
-func RichParameter(inv *serpent.Invocation, templateVersionParameter codersdk.TemplateVersionParameter) (string, error) {
+func RichParameter(inv *serpent.Invocation, templateVersionParameter codersdk.TemplateVersionParameter, defaults map[string]string) (string, error) {
```
Comment (mafredri) [inline cli/cliui/parameter.go:13]: I think this could be called overrides to better reflect what it does.

**Review by mafredri — APPROVED** [review]

**Pull request merged**